pkg/api/validate/validators.go

```diff
@@ -221,7 +221,7 @@ func validateAgentPoolProfile(app api.AgentPoolProfile, vnet *net.IPNet) (errs [
 		if app.Name != string(app.Role) {
 			errs = append(errs, fmt.Errorf("invalid properties.agentPoolProfiles[%q].name %q", app.Name, app.Name))
 		}
-		if app.Count != 2 {
+		if app.Count != 3 {
```

Comment:
Doing this will break the cli, which currently hard codes to 2.  https://github.com/Azure/azure-cli/blob/2e5c796e02a81d8b2da649065b2cf3336258650f/src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py#L2471.  Suggest allowing 2 or 3 for now and highlighting the requested change with Julien and Anubhuti.

Reply:
One more place to track our changes ;/ 
Will do